lib/vsc/mympirun/mpi/openmpi.py

```diff
@@ -85,16 +99,26 @@ def set_mpiexec_options(self):
         """Set mpiexec options"""
         super(OpenMPI, self).set_mpiexec_options()
 
-        if self.options.hybrid:
-            # specify number of processes to start per node if --hybrid is used
-            procs_per_node = self.multiplier * self.options.hybrid
-            self.mpiexec_options.add(['--map-by', 'ppr:%s:node' % procs_per_node])
+        if self.is_oversubscribed():
+            logging.debug("Allow oversubscription")
+            over = ''
         else:
-            # map MPI processes by core (default is --map-by numa),
-            # and bind to core (default is --bind-to numa' when # ranks > 2),
-            # to match default behaviour of Intel MPI;
-            # this is important for performance for OpenFOAM for example, especially on AMD Rome CPUs
-            self.mpiexec_options.add(['--map-by', 'core', '--bind-to', 'core'])
+            over = 'NO'
+
+        # make sure we start enough per node so it can fill the total_number_of_processes
+        tnop = self.total_number_of_processes()
+        un = len(self.nodes_uniq)
+        ppn = int(math.ceil(tnop / un))
```

Comment:
The variable names are a bit cryptic here, how about:

* `tnop` -> `tot_procs`
* `un` -> `uniq_nodes`
* `ppn` -> `cores_per_node`